proc/proc.go

```diff
@@ -150,35 +156,42 @@ func (p *Proc) Query() (<-chan ProcOutput, error) {
 func pollRead(
 	ctx context.Context,
 	file *os.File,
-	stream chan<- ProcOutput,
-) {
+	stream chan<- string,
+) []byte {
 	buf := make([]byte, 1024)
 	ticker := time.NewTicker(outputReadRefreshInterval)
 
 	for {
 		select {
+		case <-ctx.Done():
+			return nil
 		case <-ticker.C:
-			// ReadLoop
 			for {
 				n, err := file.Read(buf)
 				if err != nil {
 					// TODO: should we log the error somehow?
-					return
+					return nil
```

Comment:
Looking at this again is there a chance that this will exit early and miss output? For example think of a process that only prints something every few seconds you will hit an io.EOF error which will cause `pollRead` to exit but the process may still be running.